lib/joken/signer.ex

```diff
@@ -99,7 +99,8 @@ defmodule Joken.Signer do
   def create(alg, _key, _headers) when alg in @map_key_algorithms,
     do: raise(Joken.Error, :algorithm_needs_key)
 
-  def create(_, _, _), do: raise(Joken.Error, :unrecognized_algorithm)
+  def create(alg, _, _) when alg not in @algorithms,
+    do: raise(Joken.Error, :unrecognized_algorithm)
```

Comment:
I think we should guard against anything other than binary and maps with a specific error type here. 

The way it is here it will blow with a pattern match error which would be cryptic too. 

Wdyt?

Reply:
Do you mean an additional clause with those guards on the key?

Reply:
I mean, before this PR we had a catch-all clause that would match on anything. With this here we do not have a catch all anymore.

Reply:
Ah, I see. I'll change the scope of the PR to address only the actual issue then.